ch360/documents.go

```diff
@@ -166,24 +172,58 @@ func (client *DocumentsClient) Classify(ctx context.Context, documentId string,
 }
 
 func (client *DocumentsClient) Extract(ctx context.Context, documentId string, extractorName string) (*results.ExtractionResult, error) {
-	response, err := newRequest(ctx, "POST",
-		client.baseUrl+"/documents/"+documentId+"/extract/"+extractorName, nil).
-		issue(client.requestSender)
+	extractResponse := results.ExtractionResult{}
+
+	err := client.extract(ctx,
+		documentId,
+		extractorName,
+		"application/vnd.waives.resultformats.extractdata+json", &extractResponse)
 
 	if err != nil {
 		return nil, err
 	}
 
-	defer response.Body.Close()
+	return &extractResponse, nil
+}
 
-	extractResponse := results.ExtractionResult{}
-	err = json.NewDecoder(response.Body).Decode(&extractResponse)
+// ExtractForRedaction performs extracton on a document, and returns the results in a format
+// the "get redacted pdf" http endpoint will accept,
+// namely "application/vnd.waives.requestformats.redact+json".
+func (client *DocumentsClient) ExtractForRedaction(ctx context.Context,
+	documentId string,
+	extractorName string) (*results.ExtractForRedactionResult, error) {
+
+	result := results.ExtractForRedactionResult{}
```

Comment:
💪 I don't _think_ we need this as well as the `RedactedPdfRequest` type?

Reply:
one is defined as a type alias of the other :+1: 
```go
type ExtractForRedactionResult request.RedactedPdfRequest
```